src/core/server/metrics/integration_tests/server_collector.test.ts

```diff
@@ -125,7 +121,8 @@ describe.skip('ServerMetricsCollector', () => {
     );
 
     discoReq2.abort();
-    await delay(requestWaitDelay);
+    // Wait for the aborted$ event
+    await disconnectAborted$.pipe(take(1)).toPromise();
```

Comment:
Similar to my previous comment: we could use a `BehaviourSubject(counter)`. It could seem more explicit since we would check the `counter === 2` vs. this approach that looks for increments. However, it feels concurrently safer this approach.

What do you think?